test/target_azlinux_test.go

```diff
@@ -0,0 +1,145 @@
+	GreaterThan:        ">",
+	GreaterThanOrEqual: ">=",
+	LessThan:           "<",
+	LessThanOrEqual:    "<=",
+}
+
+var azlinuxTestRepoConfig = func(keyPath string) map[string]dalec.Source {
+	return map[string]dalec.Source{
+		"local.repo": {
+			Inline: &dalec.SourceInline{
+				File: &dalec.SourceInlineFile{
+					Contents: fmt.Sprintf(`[Local]
+name=Local Repository
+baseurl=file:///opt/repo
+repo_gpgcheck=1
+priority=0
+enabled=1
+gpgkey=file:///etc/pki/rpm-gpg/%s
+	`, keyPath),
+				},
+			},
+		},
+	}
+}
+
+func TestMariner2(t *testing.T) {
+	t.Parallel()
+
+	ctx := startTestSpan(baseCtx, t)
+	testLinuxDistro(ctx, t, testLinuxConfig{
+		Target: targetConfig{
+			Package:   "mariner2/rpm",
+			Container: "mariner2/container",
+			Worker:    "mariner2/worker",
+			FormatDepEqual: func(v, _ string) string {
+				return v
+			},
+			ListExpectedSignFiles: azlinuxListSignFiles("cm2"),
+		},
+		LicenseDir: "/usr/share/licenses",
+		SystemdDir: struct {
+			Units   string
+			Targets string
+		}{
+			Units:   "/usr/lib/systemd",
+			Targets: "/etc/systemd/system",
+		},
+		Worker: workerConfig{
+			ContextName:    azlinux.Mariner2WorkerContextName,
+			CreateRepo:     createYumRepo(azlinux.Mariner2Config),
+			SignRepo:       signRepoAzLinux,
+			TestRepoConfig: azlinuxTestRepoConfig,
+			Constraints:    azlinuxConstraints,
+		},
+		Release: OSRelease{
+			ID:        "mariner",
+			VersionID: "2.0",
+		},
+	})
+}
+
+func TestAzlinux3(t *testing.T) {
+	t.Parallel()
+
+	ctx := startTestSpan(baseCtx, t)
+	testLinuxDistro(ctx, t, testLinuxConfig{
+		Target: targetConfig{
+			Package:               "azlinux3/rpm",
+			Container:             "azlinux3/container",
+			Worker:                "azlinux3/worker",
+			ListExpectedSignFiles: azlinuxListSignFiles("azl3"),
+		},
+		LicenseDir: "/usr/share/licenses",
+		SystemdDir: struct {
+			Units   string
+			Targets string
+		}{
+			Units:   "/usr/lib/systemd",
+			Targets: "/etc/systemd/system",
+		},
+		Worker: workerConfig{
+			ContextName:    azlinux.Azlinux3WorkerContextName,
+			CreateRepo:     createYumRepo(azlinux.Azlinux3Config),
+			SignRepo:       signRepoAzLinux,
+			TestRepoConfig: azlinuxTestRepoConfig,
+			Constraints:    azlinuxConstraints,
+		},
+		Release: OSRelease{
+			ID:        "azurelinux",
+			VersionID: "3.0",
+		},
+	})
+}
+
+func azlinuxListSignFiles(ver string) func(*dalec.Spec, ocispecs.Platform) []string {
+	return func(spec *dalec.Spec, platform ocispecs.Platform) []string {
+		base := fmt.Sprintf("%s-%s-%s.%s", spec.Name, spec.Version, spec.Revision, ver)
+
+		var arch string
+		switch platform.Architecture {
+		case "amd64":
+			arch = "x86_64"
+		case "arm64":
+			arch = "aarch64"
+		default:
+			arch = platform.Architecture
+		}
+
+		return []string{
+			filepath.Join("SRPMS", fmt.Sprintf("%s.src.rpm", base)),
+			filepath.Join("RPMS", arch, fmt.Sprintf("%s.%s.rpm", base, arch)),
+		}
+	}
+}
+
+func signRepoAzLinux(gpgKey llb.State) llb.StateOption {
```

Comment:
Can this be made more generic and moved under `createrepo.go`?

Reply:
Perhaps as a separate refactor? This will affect the deb targets as well.

Reply:
Yes I think that would be fine!